projects/observability/src/shared/dashboard/data/graphql/topology/topology-data-source.model.ts

```diff
@@ -90,20 +100,25 @@ export class TopologyDataSourceModel extends GraphQlDataSourceModel<TopologyData
       metricSpecifications: this.getAllMetricSpecifications(this.edgeMetricsModel)
     };
 
-    return this.query<EntityTopologyGraphQlQueryHandlerService>(
-      filters => ({
+    return this.query<EntityTopologyGraphQlQueryHandlerService>(filters => {
+      const topologyFilters = this.getTopologyFilters(filters as GraphQlFieldFilter[]);
```

Comment:
we should do a check before this cast.

Reply:
What type of check? Checking any fields in the data? The  `GraphQlFIlter` interface only has one function.  and we do not know how to extract data from there.

Reply:
As in, before casting filters to `GraphQlFieldFilter`, we should check if it has properties of GraphQlFieldFilter or not. There can be Id Filters in the array that may break this method at run time.

Reply:
Okay, SO I updated the logic, Now we're checking the field filters and if it is not then we're assigning that to node filters for now.